### ISSUE TITLE
ko-bigbird 모델 학습

### ISSUE THREAD

**Issue by eunseongy** [eunseongy/NLP-class-project_Platform-Prediction-for-News-Articles#3]
###  관련 커밋
- 114445a1c7b49bae412ea61c0115a398534e0816

뉴스 본문이 매우 길기 때문에 Longformer, BigBird와 같은 Long Sequence 모델을 활용하여 학습을 진행하였으나, 성능적 한계와 처리 시간 문제로 인해 개선 전략이 필요하다고 판단됨. BigBird 기반 실험을 중심으로 chunking 전략, 하이퍼파라미터 탐색을 수행하였으며, 이후에는 긴 문장을 요약하여 한국어 특화 모델(KoELECTRA 등)에 적용하는 방향을 고려하게 됨.

### Chunking 이유

- BigBird/Longformer는 최대 4,096 token까지 입력 가능

- 뉴스 본문이 길어 max_length 제한에 걸리는 경우가 많음

- 데이터를 최대한 보존하기 위해 4000 token 단위로 chunking하여 학습함

### Chunk Filtering

마지막 chunk가 1000 tokens 이하로 남을 경우
→ 의미 정보가 적고 모델에 노이즈로 작용할 가능성이 있어 제거함

→ 결과적으로 유효 chunk만 모델 입력으로 사용하여 성능 안정화 확보

### 하이퍼 파라미터
```
batch_size = 2
lr = 1e-5
num_epochs = 10
max_length = 4096
min_ref_count = 100

```
Learning Rate 탐색

3e-5 → 2e-5 → 1e-5 로 learning rate sweep 실험 진행, batch_size=2인 것이 문제 될 수도 있어서 training_args에서 4개를 한번에 업데이트하는 아래와 같은 방법도 사용함
`gradient_accumulation_steps=4,`

실험 결과 1e-5가 가장 안정적이고 성능이 우수했음

### 장점

- 긴 문서 처리 가능

- chunking 전략과 잘 결합됨

### 단점

- 모델 용량이 매우 커서 학습 속도가 느림

- 처리 시간이 오래 걸리며 GPU 메모리 사용량도 높음

- 기대 대비 성능 향상이 제한적

→ 실험을 진행할수록 “긴 인코더라고 해서 성능이 반드시 좋은 건 아니다”라는 결론에 가까워짐

## 4. Longformer 실험 중단

학습을 진행한 후에야 해당 모델이 한국어 특화 모델이 아닌 영어 특화 모델임을 확인

한국어 데이터셋에 최적화되지 않아 성능적 이점이 없음
→ Longformer 실험은 중단하고 BigBird로 전향

##  향후 계획
 요약 후 KoELECTRA 같은 한국어 특화 모델에 입력하기

현재까지 실험을 바탕으로 다음과 같은 관찰이 있음:

긴 인코더(Longformer/BigBird)는
→ 처리 비용 대비 성능 향상이 충분하지 않음

한국어 특화 모델(KoELECTRA 계열)은
→ 훨씬 빠르고 데이터 특성에 더 잘 맞는 성능을 낼 가능성이 큼

따라서 뉴스 본문을 요약 → 한국어 특화 모델로 분류라는 새로운 접근을 시도해 보려고 함
